test/remote/gateways/remote_worldpay_test.rb

```diff
@@ -494,6 +563,36 @@ def test_successful_authorize_with_3ds_with_gateway_specific_stored_credentials
     refute first_message.params['session_id'].blank?
   end
 
+  def test_successful_purchase_with_level_two_fields
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(@level_two_data))
+    assert_success response
+    assert_equal true, response.params['ok']
+    assert_equal 'SUCCESS', response.message
+  end
+
+  def test_successful_purchase_with_level_two_fields_and_sales_tax_zero
+    @level_two_data[:sales_tax][:amount] = 0
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(@level_two_data))
+    assert_success response
+    assert_equal true, response.params['ok']
+    assert_equal 'SUCCESS', response.message
+  end
+
+  def test_successful_purchase_with_level_three_fields
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(@level_three_data))
+    assert_success response
+    assert_equal true, response.params['ok']
+    assert_equal 'SUCCESS', response.message
+  end
+
+  def test_unsuccessful_purchase_level_three_data_without_item_mastercard
+    @level_three_data[:item] = {}
+    @credit_card.brand = 'master'
+    assert response = @gateway.purchase(@amount, @credit_card, @options.merge(@level_three_data))
+    assert_failure response
+    assert_equal response.error_code, '2'
+  end
```

Comment:
💬  Comment:

Could you please add a comment on what is testing this test and what that '2' error code means.